src/lint/mod.rs

```diff
@@ -124,20 +132,38 @@ impl LinterAnalysis {
 }
 
 pub fn begin_style_check(ast: TopAst, file: String, rules: &CurrentRules) -> Result<Vec<LocalDMLError>, Error> {
-    let mut linting_errors: Vec<LocalDMLError> = vec![];
+    let mut linting_errors: Vec<DMLStyleError> = vec![];
     ast.style_check(&mut linting_errors, rules, AuxParams { depth: 0 });      
 
     // Per line checks
     let lines: Vec<&str> = file.lines().collect();
     for (row, line) in lines.iter().enumerate() {
+        rules.in2.check(&mut linting_errors, row, line);
         rules.long_lines.check(&mut linting_errors, row, line);
         rules.nsp_trailing.check(&mut linting_errors, row, line);
     }
 
     // Continuation line check
     rules.in6.check(&mut linting_errors, &lines);
 
-    Ok(linting_errors)
+    post_process_linting_errors(&mut linting_errors);
+
+    Ok(linting_errors.into_iter().map(|e| e.error).collect())
+}
+
+fn post_process_linting_errors(errors: &mut Vec<DMLStyleError>) {
+    // Collect in2 ranges
+    let in2_ranges: Vec<_> = errors.iter()
+        .filter(|style_err| style_err.rule_type == RuleType::IN2)
+        .map(|style_err| style_err.error.range)
+        .collect();
+
+    // Remove linting errors that are in in2 rows
```

Comment:
I dont know the specifics of how in2 causes conflicts with other lints, but if this is expected to be a more general problem it could be an idea to let each rule have a "priority", and report only the highest priority rule in each range.

You could also consider using some sort of span-tree structure to perform this logic on insert rather than double-iteration done here, but I would leave such optimizations until you notice the linting actually taking significant time on large devices (likely the time will always be dominated by the parser anyway)